### PR TITLE
docs(ui5-dialog): add samples of ui5-dialog with ui5-bar

### LINKED ISSUE #6638: ui5-bar: padding/border issues when slotted as header/footer of Dialog or ResponsivePopover
### **Bug Description**
The UI5-bar when slotted as the header or footer of a Dialog or ResponsivePopover has an extra top border on the footer, and too much padding on the left/right sides of both header and footer.

### **Expected Behavior**
The Dialog or Bar should account for when using this combination of components. Do not render a second border/box-shadow on the footer, and don't add extra left/right padding. When there is text content in the Dialog and in the startContent of the Bar, that text content should align left perfectly.

### **Steps to Reproduce**
1. https://codesandbox.io/s/ui5-webcomponents-forked-2sflyw?file=/index.html
2. See the double border on the footer and the extra left/right padding

### **Isolated Example**
https://codesandbox.io/s/ui5-webcomponents-forked-2sflyw?file=/index.html

### **Context**
- UI5 Web Components version: latest
- OS/Platform: any
- Browser: any
- Affected component: bar, dialog

### **Log Output / Stack Trace / Screenshots**
```
{...}
```
### **Priority**
- [ ] Low
- [x] Medium
- [ ] High
- [ ] Very High

The priority indicates the severity of the issue. To set the appropriate priority consider the following criteria:
- Breaks entire application or system - High or Very High
- Accessibility issue - Medium or High
- Functional issue - Medium or High
- Visual issue - Low or Medium

**Note:** The priority might be re-evaluated by the issue processor.

### **Stakeholder Info** (if applicable)

- Organization: {...}
- Business impact: {...}

### PR DESCRIPTION
**Pull request by LidiyaGeorgieva**
Custom styles for removing double side paddings when
ui5-bar is used in the header and/or in the footer of the ui5-dialog
added in the documentation of the ui5-dialog.

Fixes #6638

### CODE REVIEW

File: packages/playground/_stories/main/Dialog/Dialog.stories.ts [hunk 1]
```diff
@@ -120,6 +120,26 @@ FioriDialog.args = {
 </div>`,
 };
 
+export const BarInDialog = Template.bind({});
+BarInDialog.storyName = "Bar in Header/Footer";
+BarInDialog.args = {
+	header: '<ui5-bar slot="header" design="Header"> <ui5-text slot="startContent">Bar used in Header and Footer</ui5-text> <ui5-button class="dialogCloser" design="Transparent" id="closeDialogButton" slot="endContent" icon="decline"/> </ui5-bar>',
```
Comment (gmkv) [inline packages/playground/_stories/main/Dialog/Dialog.stories.ts:126]: You can give this HTML some proper indentation by using a template literal instead.
There's also no such component like ui5-text, perhaps you meant to use a ui5-title?
And custom elements can't be self-closing. It is a requirement for them to have a separate closing tag.

File: packages/playground/_stories/main/Dialog/Dialog.stories.ts [hunk 2]
```diff
@@ -120,6 +120,26 @@ FioriDialog.args = {
...
+	default: "<p>Adding styles for the parts to remove the default Dialog's paddings when ui5-bar is used inside Header and Footer</p>",
+	footer: `<ui5-bar slot="footer" design="Footer">
+	<ui5-button class="dialogCloser" design="Emphasized" slot="endContent">OK</ui5-button>
```
Comment (gmkv) [inline packages/playground/_stories/main/Dialog/Dialog.stories.ts:129]: Add an inline style here or an extra style below to give all footer buttons a min-width of 4rem. This comes from the Fiori Guidelines (or VD spec). See the usage on line 118.

**Review by gmkv — CHANGES_REQUESTED** [review]

**Comment by gmkv** [comment]
Also, rewrite the title of the PR to a sentence which explains more on what this commit does.

File: packages/playground/_stories/main/Dialog/Dialog.stories.ts [hunk 3]
```diff
@@ -120,6 +120,29 @@ FioriDialog.args = {
 </div>`,
 };
 
+export const BarInDialog = Template.bind({});
+BarInDialog.storyName = "Bar in Header/Footer";
+BarInDialog.args = {
+	header: `<ui5-bar slot="header" design="Header">
+				<ui5-title level="H5" slot="startContent">Bar used in Header and Footer</ui5-title>
+				<ui5-button class="dialogCloser" design="Transparent" id="closeDialogButton" slot="endContent" icon="decline"></ui5-button>
+			</ui5-bar>`,
```
Comment (gmkv) [inline packages/playground/_stories/main/Dialog/Dialog.stories.ts:129]: this indentation should be corrected to so it looks right in the Storybook page:
image


We can't make the indentation look ok both in Storybook and in TS, so we choose to make it look correct only for Storybook

**Review by gmkv — CHANGES_REQUESTED** [review]

**Review by gmkv — APPROVED** [review]

File: packages/fiori/test/pages/DialogWithBar.html [hunk 4]
```diff
@@ -11,11 +11,30 @@
...
+		padding-inline: 0;
+	}
+</style>
+
 <body class="bar1auto">
 	<ui5-button id="dialog-bar-open">Dialog with Bar in Footer</ui5-button>
-	<ui5-dialog open id="dialog-bar" header-text="Using bar with design=Footer as footer">
+	<ui5-dialog open resizable id="dialog-bar" header-text="Using bar with design=Footer as footer">
+		<ui5-bar slot="header" design="Header">
+			<ui5-title level="H5" slot="startContent">Sample Dialog With Bar</ui5-title>
```
Comment (BorisDafov) [inline packages/fiori/test/pages/DialogWithBar.html:25]: no need to capitalize "with":
Sample Dialog with Bar

File: packages/main/test/samples/Dialog.sample.html [hunk 5]
```diff
@@ -198,6 +198,84 @@ <h3>Dialog with SAP Fiori Styled Footer</h3>
...
+
+		dialogCloser42.addEventListener("click", function() {
+			dialog4.close();
+		});
+	</script>
+
+	<pre class="prettyprint lang-html"><xmp>
+		<ui5-dialog id="with-bar-dialog">
+			<ui5-bar slot="header" design="Header">
+				<ui5-title level="H5" slot="startContent">Sample Dialog With Bar</ui5-title>
```
Comment (BorisDafov) [inline packages/main/test/samples/Dialog.sample.html:247]: with

File: packages/main/test/samples/Dialog.sample.html [hunk 6]
```diff
@@ -198,6 +198,84 @@ <h3>Dialog with SAP Fiori Styled Footer</h3>
...
 </section>
 
+<section>
+	<h3>Dialog with Bar in the Header and in the Footer</h3>
+	<div class="snippet">
+		<ui5-button id="with-bar">Open dialog with ui5-bar</ui5-button>
+
+		<ui5-dialog id="with-bar-dialog">
+			<ui5-bar slot="header" design="Header">
+				<ui5-title level="H5" slot="startContent">Sample Dialog With Bar</ui5-title>
```
Comment (BorisDafov) [inline packages/main/test/samples/Dialog.sample.html:208]: with

**Review by BorisDafov — CHANGES_REQUESTED** [review]
minor

**Review by BorisDafov — APPROVED** [review]

**Review by gmkv — APPROVED** [review]

**Pull request merged**